sanic/http/http1.py

```diff
@@ -132,15 +132,18 @@ async def http1(self):
 
                 if self.stage is Stage.RESPONSE:
                     await self.response.send(end_stream=True)
-            except CancelledError:
+            except (CancelledError, RequestCancelled) as exc:
                 # Write an appropriate response before exiting
                 if not self.protocol.transport:
                     logger.info(
                         f"Request: {self.request.method} {self.request.url} "
                         "stopped. Transport is closed."
                     )
                     return
-                e = self.exception or ServiceUnavailable("Cancelled")
+                if self.protocol.conn_info.lost:
+                    e = RequestCancelled
+                else:
+                    e = self.exception or exc
```

Comment:
Should that be RequestCancelled(...) rather than bare type? Might also prefer expression conditional:
```python
e = RequestCancelled() if self.protocol.conn_info.lost else self.exception or exc
```

Reply:
I usually try to avoid multiple branches in a single expression. But that's fine with me.